### ISSUE TITLE
A nonreproducible crash

### ISSUE THREAD

**Issue by tsujan** [QupZilla/qupzilla#2498]
This is release v2.2.1 compiled against Qt 5.9.2. I can't tell why the crash happened but this is the backtrace:
```
#0  0x00007f4266f288a0 in raise () at /usr/lib/libc.so.6
#1  0x00007f4266f29f09 in abort () at /usr/lib/libc.so.6
#2  0x00007f4266f6b517 in __libc_message () at /usr/lib/libc.so.6
#3  0x00007f4266f71c84 in malloc_printerr () at /usr/lib/libc.so.6
#4  0x00007f4266f76364 in _int_realloc () at /usr/lib/libc.so.6
#5  0x00007f4266f7722e in realloc () at /usr/lib/libc.so.6
#6  0x00007f42678f684b in QArrayData::reallocateUnaligned(QArrayData*, unsigned long, unsigned long, QFlags<QArrayData::AllocationOption>) () at /usr/lib/libQt5Core.so.5
#7  0x00007f42678f8f53 in QByteArray::reallocData(unsigned int, QFlags<QArrayData::AllocationOption>) ()
    at /usr/lib/libQt5Core.so.5
#8  0x00007f42678f912b in QByteArray::resize(int) () at /usr/lib/libQt5Core.so.5
#9  0x00007f4267b3ce29 in  () at /usr/lib/libQt5Core.so.5
#10 0x00007f4267b3d195 in  () at /usr/lib/libQt5Core.so.5
#11 0x00007f4267972a67 in QString::toLocal8Bit_helper(QChar const*, int) () at /usr/lib/libQt5Core.so.5
#12 0x000055ac64eafdeb in  ()
#13 0x00007f4266f28920 in <signal handler called> () at /usr/lib/libc.so.6
#14 0x00007f426704ba95 in __memmove_avx_unaligned_erms () at /usr/lib/libc.so.6
#15 0x00007f4267947d41 in QListData::remove(int) () at /usr/lib/libQt5Core.so.5
#16 0x00007f423948a691 in  () at /usr/lib/qt/plugins/sqldrivers/libqsqlite.so
#17 0x00007f423948789c in  () at /usr/lib/qt/plugins/sqldrivers/libqsqlite.so
#18 0x00007f4239487909 in  () at /usr/lib/qt/plugins/sqldrivers/libqsqlite.so
#19 0x00007f425e671262 in  () at /usr/lib/libQt5Sql.so.5
#20 0x00007f425e67162a in QSqlQuery::~QSqlQuery() () at /usr/lib/libQt5Sql.so.5
#21 0x00007f4268a09c02 in IconProvider::imageForUrl(QUrl const&, bool) () at /usr/lib/libQupZilla.so.2
#22 0x00007f4268a0a5f0 in IconProvider::iconForUrl(QUrl const&, bool) () at /usr/lib/libQupZilla.so.2
#23 0x00007f42688e74c7 in BookmarkItem::icon() () at /usr/lib/libQupZilla.so.2
#24 0x00007f42689036ae in BookmarksMenu::menuAboutToShow() () at /usr/lib/libQupZilla.so.2
#25 0x00007f4268a3ce0c in  () at /usr/lib/libQupZilla.so.2
#26 0x00007f4267b08c16 in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#27 0x00007f4268208f4d in QMenu::popup(QPoint const&, QAction*) () at /usr/lib/libQt5Widgets.so.5
#28 0x00007f426820ba30 in QMenu::internalDelayedPopup() () at /usr/lib/libQt5Widgets.so.5
#29 0x00007f4267b0961b in QObject::event(QEvent*) () at /usr/lib/libQt5Core.so.5
#30 0x00007f42680c713c in QWidget::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#31 0x00007f4268210a9c in QMenu::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#32 0x00007f4268086f2c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#33 0x00007f426808ea06 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#34 0x00007f4267ad9060 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#35 0x00007f4267b33118 in QTimerInfoList::activateTimers() () at /usr/lib/libQt5Core.so.5
#36 0x00007f4267b3394a in  () at /usr/lib/libQt5Core.so.5
#37 0x00007f425c86a270 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#38 0x00007f425c86bf69 in  () at /usr/lib/libglib-2.0.so.0
#39 0x00007f425c86bfae in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#40 0x00007f4267b33cd1 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /usr/lib/libQt5Core.so.5
#41 0x00007f4267ad748b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#42 0x00007f4267ae04d8 in QCoreApplication::exec() () at /usr/lib/libQt5Core.so.5
#43 0x000055ac64eaf7aa in  ()
#44 0x00007f4266f14f6a in __libc_start_main () at /usr/lib/libc.so.6
#45 0x000055ac64eaf96a in  ()
```

**Comment by ymuv** [comment]
I have same problem.

**Comment by ymuv** [comment]
I have same problem: 
```
#0  0x00007fffee266f30 in raise () from /lib64/libc.so.6
#1  0x00007fffee268bda in abort () from /lib64/libc.so.6
#2  0x00007fffee2a9a30 in __libc_message () from /lib64/libc.so.6
#3  0x00007fffee2b169e in malloc_printerr () from /lib64/libc.so.6
#4  0x00007fffee2b3bf0 in _int_malloc () from /lib64/libc.so.6
#5  0x00007fffee2b47c1 in _int_realloc () from /lib64/libc.so.6
#6  0x00007fffee2b5df6 in realloc () from /lib64/libc.so.6
#7  0x00007fffeeca6363 in QArrayData::reallocateUnaligned(QArrayData*, unsigned long, unsigned long, QFlags<QArrayData::AllocationOption>) () from /usr/lib64/libQt5Core.so.5
#8  0x00007fffeeca826c in QByteArray::reallocData(unsigned int, QFlags<QArrayData::AllocationOption>) ()
   from /usr/lib64/libQt5Core.so.5
#9  0x00007fffeeca8548 in QByteArray::resize(int) () from /usr/lib64/libQt5Core.so.5
#10 0x00007fffeed903c5 in QBuffer::writeData(char const*, long long) () from /usr/lib64/libQt5Core.so.5
#11 0x00007fffeedafe50 in QIODevice::write(char const*, long long) () from /usr/lib64/libQt5Core.so.5
#12 0x00007fffeed919fa in QDataStream::writeRawData(char const*, int) () from /usr/lib64/libQt5Core.so.5
#13 0x00007fffeed91ab9 in QDataStream::writeBytes(char const*, unsigned int) () from /usr/lib64/libQt5Core.so.5
#14 0x00007fffeed34ee5 in operator<<(QDataStream&, QString const&) () from /usr/lib64/libQt5Core.so.5
#15 0x0000000000659c1a in QtPrivate::writeAssociativeContainer<QMap<QString, QVariant> > (s=..., c=...)
    at /usr/include/qt5/QtCore/qdatastream.h:316
#16 0x000000000065987b in operator<< <QString, QVariant> (s=..., map=...) at /usr/include/qt5/QtCore/qdatastream.h:441
#17 0x00000000006740c1 in operator<< (out=..., send=...) at /home/y/vaglive/cameraAlerts_new/common/private/CRequestData.cpp:6
#18 0x0000000000648ff9 in CDeSerializer::ClientSerialize (data=..., outStr=...)

```

**Comment by tsujan** [comment]
Here, it hasn't happened since v2.2.2. Fixed?

**Comment by ymuv** [comment]
Sorry, I push comment to another application.

**Comment by tsujan** [comment]
> Sorry, I push comment to another application.

You could delete the comment.

**Comment by nowrep** [comment]
I received same report yesterday, so I don't think it is fixed.
Actually it should have been fixed in 2.2.1 and it should have only occur with Qt 5.10, but you reported it against 2.2.1 + Qt 5.9...

**Comment by tsujan** [comment]
> Actually it should have been fixed in 2.2.1 and it should have only occur with Qt 5.10, but you reported it against 2.2.1 + Qt 5.9...

Yes, it happened with those versions twice but, since v2.2.1, I haven't encountered any crash. If not fixed, it might be random and rare; I'll add a comment if it happens again.
